### PR TITLE
Implement session

### PR DESCRIPTION
**Pull request by GingerBoy14**

### CODE REVIEW

File: src/app/services/Firestore/getData.js [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+      }
+      result = await result.data()
+      return result
+    } catch (e) {
+      console.log('data', e)
+    }
   }
 
   const res = await getCollectionRef(collection).get()
+  console.log(res)
```
Comment (AlexFry123) [inline src/app/services/Firestore/getData.js:25]: Remove console logs

File: src/app/context/SessionContext/rootReducer.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import * as type from './types'
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case type.LOGIN_USER: {
+      console.log(action.payload)
```
Comment (olenashevchuk) [inline src/app/context/SessionContext/rootReducer.js:6]: delete console.log

File: src/app/domains/Session/routes/SessionLogin/SessionLogin.layout.js [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
   return (
     <Row v="center" height="100%">
       <Col>
         <Paper>
           <Box p={2}>
-            <SessionSimpleView />
+            <SessionSimpleView login={login} />
           </Box>
         </Paper>
       </Col>
```
Comment (AlexFry123) [inline src/app/domains/Session/routes/SessionLogin/SessionLogin.layout.js:22]: Line 27 add prop types (for login prop)

File: src/app/components/RejectLogin/RejectLogin.js [hunk 4]
```diff
@@ -1,23 +1,32 @@
+import { useEffect } from 'react'
 import { Col, Container, Row } from '@qonsoll/react-design'
 import Typography from '@material-ui/core/Typography'
 import Button from '@material-ui/core/Button'
```
Comment (AlexFry123) [inline src/app/components/RejectLogin/RejectLogin.js:4]: Group imports from one package/lib

File: src/app/components/RejectLogin/RejectLogin.js [hunk 5]
```diff
@@ -1,23 +1,32 @@
...
 
 const RejectLogin = () => {
   let history = useHistory()
   const redirect = () => history.push(ROUTES_PATHS.LOGIN)
-
+  //logout rejected user after 10 seconds
+  useEffect(() => {
+    setTimeout(() => auth.signOut(), 10000)
+    return () => {
+      console.log('logout')
```
Comment (AlexFry123) [inline src/app/components/RejectLogin/RejectLogin.js]: Remove console logs

File: src/app/components/RejectLogin/RejectLogin.js [hunk 6]
```diff
@@ -1,23 +1,32 @@
...
+      auth.signOut()
```
Comment (AlexFry123) [inline src/app/components/RejectLogin/RejectLogin.js]: Why signOut twice? inside component mount and after unmount

File: src/app/hooks/useAuthListener.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
+import md5 from 'md5'
+import { useAuthState } from 'react-firebase-hooks/auth'
+import { auth } from 'app/services/Auth'
+import { getData } from 'app/services/Firestore'
+import { COLLECTIONS, ROUTES_PATHS } from 'app/constants'
+//HELPERS
+const loginUser = (user) => {}
```
Comment (AlexFry123) [inline src/app/hooks/useAuthListener.js]: 👀

File: src/app/hooks/useAuthListener.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  const [reqError, setReqError] = useState(false)
+  const [load, setLoad] = useState(true)
+  let history = useHistory()
+  console.log(user, loading, error)
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        setLoad(true)
+        const res = await getData(COLLECTIONS.USERS, md5(user.email))
+        console.log(res)
```
Comment (AlexFry123) [inline src/app/hooks/useAuthListener.js]: Remove console log

File: src/app/hooks/useAuthListener.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      } catch (e) {
+        if (e.message === 'document not exist.') {
+          setLoad(false)
+          return setReqError(true)
+        }
```
Comment (AlexFry123) [inline src/app/hooks/useAuthListener.js:59]: Add at least exception logging for errors with message that isn't equal to "document doesn't exist"

File: src/app/hooks/useAuthListener.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      }
+      setLoad(false)
+    }
+    if (user === null) {
+      history.replace(ROUTES_PATHS.LOGIN)
+      !loading && setLoad(false)
+    }
+    !loading && fetchUser()
+  }, [user, loading])
+  console.log(reqError)
```
Comment (AlexFry123) [inline src/app/hooks/useAuthListener.js]: Remove console log

**Review by AlexFry123 — APPROVED** [review]

**Review by AlexFry123 — APPROVED** [review]

**Review by okrdima — APPROVED** [review]

**Pull request merged**